src/core/ActivityInfoTree.cpp

```diff
@@ -272,4 +275,73 @@ void ActivityInfoTree::init()
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
 }
+void ActivityInfoTree::beginsearch()
+{
+
+     m_menuTree = m_searched;
+     emit menuTreeChanged();
+     m_mainWindow = ApplicationInfo::getWindow()->findChild<QObject*>("menu");
```

Comment:
You should do like for the filterByTag, call the method display(QString) in qml.